### PR TITLE
Add Player Warden Warning API

### PR DESCRIPTION
**Pull request by dawon**
I am still not sure whether to mention specific values (for warning levels 0-4) as those might change. So I tried to avoid them, if it is preferred to have them in the docs I will modify it...

### CODE REVIEW

File: patches/api/0402-Add-Player-Warden-Warning-API.patch [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+index ecb8b4cf48c7d6151ffec92bc6855d1fc57a2b51..0ef887de348cd91431438deb9bc2c25136b1ee5b 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -2844,6 +2844,59 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+      * @param silent whether sound should be silenced
+      */
+     void showElderGuardian(boolean silent);
++
++    /**
++     * Returns the player's cooldown till next Warden warning can occur in ticks.
```
Comment (kennytv) [inline patches/api/0402-Add-Player-Warden-Warning-API.patch]:
```suggestion
+     * Returns the player's cooldown in ticks until the next Warden warning can occur.
```

and so on with "till"

**Review by kennytv — CHANGES_REQUESTED** [review]

File: patches/api/0402-Add-Player-Warden-Warning-API.patch (hunk 1 above)
Comment (dawon) [reply patches/api/0402-Add-Player-Warden-Warning-API.patch]: should be rebased & fixed, thank you!

**Comment by dawon** [comment]
rebased

File: build-data/paper.at [hunk 2]
```diff
@@ -360,3 +360,8 @@ public net.minecraft.world.entity.projectile.Projectile leftOwner
 # Teleport API
 public net.minecraft.server.network.ServerGamePacketListenerImpl internalTeleport(DDDFFLjava/util/Set;Z)V
 
+# Player Warden Warning API
```
Comment (Owen1212055) [inline build-data/paper.at]: Access transformers are kind of being phased out for uses like this. In general, just modify the fields in the patch .

Comment (dawon) [reply build-data/paper.at]: Thought the trend was to get everything to the transformers... :-D nvm, resolved and rebased

**Review by Owen1212055 — APPROVED** [review]

**Comment by Owen1212055** [comment]
I’m so sorry about this, but you can now re add the access transformers except have them in the patch header. See the paper contributing guide. 😅

**Comment by MiniDigger** [comment]
this PR caused an internal discussion about ATs which prompted a new feature in paperweight that allows us to use ATs again without stuff becoming messy, so thanks for that, lol
 https://github.com/PaperMC/paperweight/pull/171
https://github.com/PaperMC/Paper/pull/8573

**Comment by dawon** [comment]
yeah, I saw the change, so kinda expected that and planned on doing it, but did not get to it... I thought the old spigot way was not as good as ATs, this one is perfect! fixed & rebased

**Review by kennytv — APPROVED** [review]

**Review by Owen1212055 — APPROVED** [review]

**Pull request merged**